dis_snek/api/events/processors/role_events.py

```diff
@@ -35,4 +40,16 @@ async def _on_raw_guild_role_update(self, event: "RawGatewayEvent") -> None:
 
     @Processor.define()
     async def _on_raw_guild_role_delete(self, event: "RawGatewayEvent") -> None:
-        self.dispatch(events.RoleDelete(event.data.get("guild_id"), event.data.get("role_id")))
+        g_id = int(event.data.get("guild_id"))
+        r_id = int(event.data.get("role_id"))
+
+        guild = self.cache.guild_cache.get(g_id)
+        role = self.cache.role_cache.pop(r_id, MISSING)
+
+        guild._role_ids.discard(r_id)
+
+        role_members = (member for member in guild.members if member.has_role(r_id))
+        for member in role_members:
+            member._role_ids.remove(r_id)
```

Comment:
Imo this should be error-handled. If *somehow* the ID isnt already in the list, this will fail with a `ValueError`. Better to handle that edge case

Reply:
I don't quite agree because `member.has_role` checks if `role id` is in `member._role_ids`, and then it is removed from `member._role_ids` in for cycle. So there is no way for a real edge case here, as both segments of code check and assume the same thing. Given that there is nothing async involved, nothing sane modify data mid-cycle either

Reply:
Reasonable point, resolving this thread